### PR TITLE
fix(auth): Improve OAuth credential detection and startup warning

### LINKED ISSUE #721: [Bug]: Startup warning ignores Claude OAuth credentials
### Operating System

Linux (WSL2)

### Run Mode

Docker

### App Version

v0.14.0rc

### Bug Description

When running Automaker via Docker with Claude OAuth credentials mounted (from `~/.claude/.credentials.json`), the server startup still shows:

```
WARN  [Server]
╔═════════════════════════════════════════════════════════════════════╗
║  ⚠️  WARNING: No Claude authentication configured                   ║
╠═════════════════════════════════════════════════════════════════════╣
║                                                                     ║
║  The Claude Agent SDK requires authentication to function.          ║
║                                                                     ║
║  Set your Anthropic API key:                                        ║
║    export ANTHROPIC_API_KEY="sk-ant-..."                            ║
║                                                                     ║
║  Or use the setup wizard in Settings to configure authentication.   ║
║                                                                     ║
╚═════════════════════════════════════════════════════════════════════╝
```

This warning appears even though OAuth credentials are properly mounted and accessible inside the container.

### Root Cause

The startup check in `apps/server/src/index.ts:121-148` only checks for `ANTHROPIC_API_KEY` environment variable:

```typescript
const hasAnthropicKey = !!process.env.ANTHROPIC_API_KEY;

if (!hasAnthropicKey) {
  // Shows warning
}
```

However, the `getClaudeStatus()` function in `apps/server/src/routes/setup/get-claude-status.ts:102-172` has comprehensive OAuth detection logic that:
- Checks for `.credentials.json` with OAuth tokens via `getClaudeAuthIndicators()`
- Detects CLI authentication indicators
- Checks for recent activity

The startup warning should use the same detection logic (or a simplified sync version) to also detect OAuth credentials before displaying the warning.

### Steps to Reproduce

1. Authenticate with Claude CLI (`claude login`) on host
2. Configure `docker-compose.override.yml` to mount `~/.claude` directory
3. Run `docker-compose up`
4. Observe the warning despite OAuth credentials being available

### Expected Behavior

The startup check should detect OAuth credentials from `~/.claude/.credentials.json` and NOT show the warning when valid OAuth credentials are present.

### Actual Behavior

Warning is always shown when `ANTHROPIC_API_KEY` is not set, even if OAuth credentials are configured.

### Suggested Fix

The startup check should call `getClaudeAuthIndicators()` from `@automaker/platform` (or a simplified sync check) to detect OAuth credentials. Something like:

```typescript
import { getClaudeAuthIndicators } from '@automaker/platform';

// At startup
const hasAnthropicKey = !!process.env.ANTHROPIC_API_KEY;
const indicators = await getClaudeAuthIndicators();
const hasOAuth = indicators.hasCredentialsFile && indicators.credentials?.hasOAuthToken;

if (!hasAnthropicKey && !hasOAuth) {
  // Show warning
} else if (hasOAuth && !hasAnthropicKey) {
  logger.info('✓ Claude OAuth credentials detected');
}
```

### Screenshots

_No response_

### Relevant Logs

```shell
automaker-server  | WARN  [Server]
automaker-server  | ╔═════════════════════════════════════════════════════════════════════╗
automaker-server  | ║  ⚠️  WARNING: No Claude authentication configured                   ║
...
```

### Additional Context

The OAuth credentials are accessible inside the container (verified with `ls -la` showing `.credentials.json`).

### Checklist

- [x] I have searched existing issues to ensure this bug hasn't been reported already
- [x] I have provided all required information above

### PR DESCRIPTION
**Pull request by Shironex**
- Enhanced getClaudeAuthIndicators() to return detailed check information including file paths checked and specific error details for debugging
- Added debug logging to server startup credential detection for easier troubleshooting in Docker environments
- Show paths that were checked in the warning message to help users debug mount issues
- Added support for CLAUDE_CODE_OAUTH_TOKEN environment variable
- Return authType in verify-claude-auth response to distinguish between OAuth and CLI authentication methods
- Updated UI to show specific success messages for Claude Code subscription vs generic CLI auth
- Added Docker troubleshooting tips to sandbox risk dialog
- Added comprehensive unit tests for OAuth credential detection scenarios

Closes #721

## Summary by CodeRabbit

* **New Features**
  * OAuth token auth supported and auth method now reported (api_key, oauth, cli).

* **Improvements**
  * Richer auth diagnostics with per-path existence/readability checks and clearer success/warning messages per auth type.
  * UI updates: adaptive thinking option, model-aware thinking selector, OAuth-specific success text, and Docker troubleshooting guidance.
  * New Codex model entry: GPT‑5.3‑Codex; Claude default moved to Opus 4.6.

* **Tests**
  * Extensive credential-detection and edge-case test suite.

* **Chores**
  * Updated ignore entries and documentation model references.

### CODE REVIEW

File: apps/server/src/index.ts [hunk 1]
```diff
@@ -158,6 +226,33 @@ const BOX_CONTENT_WIDTH = 67;
...
+    }
+    for (const credFile of cliAuthIndicators.checks.credentialFiles) {
+      pathsChecked.push(`Credentials: ${credFile.path}`);
+    }
+
+    if (pathsChecked.length > 0) {
+      pathsCheckedInfo = `
+║                                                                     ║
+║  ${'Paths checked:'.padEnd(BOX_CONTENT_WIDTH)}║
+${pathsChecked.map((p) => `║    ${p.substring(0, BOX_CONTENT_WIDTH - 2).padEnd(BOX_CONTENT_WIDTH - 2)}  ║`).join('\n')}`;
```
Comment (Shironex) [inline apps/server/src/index.ts]: > [!WARNING]
> **Major - Logic:** Warning box path lines are 2 characters too wide, breaking ASCII box alignment

The path lines in the warning box have incorrect width calculation. The box interior is 69 characters wide (71 total - 2 for ║ borders). Each path line uses: 4 leading spaces + content.padEnd(65) + 2 trailing spaces = 71 interior characters, which is 2 more than the 69-char box interior. This causes the right border ║ to be misaligned, breaking the visual box formatting in the console warning. The 'Paths checked:' header line just above uses the correct `BOX_CONTENT_WIDTH` (67) with `║  ` prefix (2 spaces), which totals 69 interior chars correctly.

**Problematic code:**
```typescript
${pathsChecked.map((p) => `║    ${p.substring(0, BOX_CONTENT_WIDTH - 2).padEnd(BOX_CONTENT_WIDTH - 2)}  ║`).join('\n')}
```

**Suggested fix:**
```diff
// Change BOX_CONTENT_WIDTH - 2 to BOX_CONTENT_WIDTH - 4 to account for
// the 4 leading spaces + 2 trailing spaces (6 chars overhead vs 4 for standard lines)
${pathsChecked.map((p) => `║    ${p.substring(0, BOX_CONTENT_WIDTH - 4).padEnd(BOX_CONTENT_WIDTH - 4)}  ║`).join('\n')}
```

File: apps/server/src/routes/setup/routes/verify-claude-auth.ts [hunk 2]
```diff
@@ -320,9 +320,28 @@ export function createVerifyClaudeAuthHandler() {
...
 
+      // Determine specific auth type for success messages
+      let authType: 'oauth' | 'api_key' | 'cli' | undefined;
+      if (authenticated) {
+        if (authMethod === 'api_key') {
+          authType = 'api_key';
+        } else if (authMethod === 'cli') {
+          // Check if CLI auth is via OAuth (Claude Code subscription) or generic CLI
+          // OAuth tokens are stored in the credentials file by the Claude CLI
+          const { getClaudeAuthIndicators } = await import('@automaker/platform');
```
Comment (Shironex) [inline apps/server/src/routes/setup/routes/verify-claude-auth.ts]: > [!IMPORTANT]
> **Minor - Logic:** getClaudeAuthIndicators failure in verify handler causes 500 error despite successful auth

If `getClaudeAuthIndicators()` throws (e.g., file system error), the entire request handler's outer catch block returns a 500 error, losing the already-determined authentication success result. Since this call is purely for determining the auth sub-type (oauth vs cli) and is not critical to the auth verification itself, it should be wrapped in its own try/catch so that a failure here gracefully falls back to 'cli' auth type rather than crashing the endpoint.

**Problematic code:**
```typescript
const { getClaudeAuthIndicators } = await import('@automaker/platform');
const indicators = await getClaudeAuthIndicators();
```

**Suggested fix:**
```diff
let authType: 'oauth' | 'api_key' | 'cli' | undefined;
if (authenticated) {
  if (authMethod === 'api_key') {
    authType = 'api_key';
  } else if (authMethod === 'cli') {
    try {
      const { getClaudeAuthIndicators } = await import('@automaker/platform');
      const indicators = await getClaudeAuthIndicators();
      authType = indicators.credentials?.hasOAuthToken ? 'oauth' : 'cli';
    } catch {
      // Fall back to generic CLI if credential check fails
      authType = 'cli';
    }
  }
}
```

Comment (Shironex) [inline apps/server/src/routes/setup/routes/verify-claude-auth.ts]: > [!IMPORTANT]
> **Minor - Performance:** Unnecessary dynamic import for @automaker/platform in request handler

Using dynamic `import()` inside a request handler when the module could be statically imported at the top of the file. `@automaker/platform` is already imported elsewhere in the codebase and the module is always available. While the runtime cost of a cached dynamic import is minimal, static imports are the established pattern in this file (see line 10: `import { ... } from '../../../lib/auth-utils.js'`) and make dependencies clearer.

**Problematic code:**
```typescript
const { getClaudeAuthIndicators } = await import('@automaker/platform');
```

**Suggested fix:**
```diff
// At the top of the file, add to existing imports:
import { getClaudeAuthIndicators } from '@automaker/platform';

// Then in the handler, use directly:
const indicators = await getClaudeAuthIndicators();
```

File: apps/server/src/index.ts [hunk 3]
```diff
@@ -121,21 +121,89 @@ const BOX_CONTENT_WIDTH = 67;
...
+    // Log detailed credential detection results
+    logger.debug('[CREDENTIAL_CHECK] Claude CLI auth indicators:', {
+      hasCredentialsFile: indicators.hasCredentialsFile,
+      hasSettingsFile: indicators.hasSettingsFile,
+      hasStatsCacheWithActivity: indicators.hasStatsCacheWithActivity,
+      hasProjectsSessions: indicators.hasProjectsSessions,
+      credentials: indicators.credentials,
+    });
+
+    logger.debug('[CREDENTIAL_CHECK] File check details:', {
```
Comment (Shironex) [inline apps/server/src/index.ts]: > [!NOTE]
> **Nit - Style:** Verbose debug logging manually reconstructs objects that could be logged directly

The debug logging manually destructures every field from the check objects to build identical objects. Since these objects are already structured data suitable for logging, the entire `indicators.checks` object can be logged directly, which is more maintainable and won't drift if fields are added/removed from the interfaces.

**Problematic code:**
```typescript
logger.debug('[CREDENTIAL_CHECK] File check details:', {
  settingsFile: {
    path: indicators.checks.settingsFile.path,
    exists: indicators.checks.settingsFile.exists,
    readable: indicators.checks.settingsFile.readable,
    error: indicators.checks.settingsFile.error,
  },
  ...
```

**Suggested fix:**
```diff
// Instead of manually rebuilding each check object, log directly:
logger.debug('[CREDENTIAL_CHECK] File check details:', indicators.checks);
```

File: apps/server/src/index.ts [hunk 4]
```diff
@@ -121,21 +121,89 @@ const BOX_CONTENT_WIDTH = 67;
...
+  // Store indicators outside the try block so we can use them in the warning message
+  let cliAuthIndicators: Awaited<ReturnType<typeof getClaudeAuthIndicators>> | null = null;
+
   try {
-    const indicators = await getClaudeAuthIndicators();
+    cliAuthIndicators = await getClaudeAuthIndicators();
+    const indicators = cliAuthIndicators;
+
+    // Log detailed credential detection results
+    logger.debug('[CREDENTIAL_CHECK] Claude CLI auth indicators:', {
```
Comment (Shironex) [inline apps/server/src/index.ts]: > [!NOTE]
> **Nit - Style:** Redundant field-by-field destructuring in debug log

Similar to the file check details, this debug log manually picks fields from the indicators object. The top-level fields (minus `checks`) could be destructured more concisely. However, this is a minor style preference - the explicit listing does make it clear exactly what's logged at debug level.

**Problematic code:**
```typescript
logger.debug('[CREDENTIAL_CHECK] Claude CLI auth indicators:', {
  hasCredentialsFile: indicators.hasCredentialsFile,
  hasSettingsFile: indicators.hasSettingsFile,
  hasStatsCacheWithActivity: indicators.hasStatsCacheWithActivity,
  hasProjectsSessions: indicators.hasProjectsSessions,
  credentials: indicators.credentials,
});
```

**Suggested fix:**
```diff
// Option: destructure to exclude the verbose `checks` object
const { checks, ...indicatorSummary } = indicators;
logger.debug('[CREDENTIAL_CHECK] Claude CLI auth indicators:', indicatorSummary);
```

File: apps/server/src/index.ts [hunk 5]
```diff
@@ -158,6 +226,33 @@ const BOX_CONTENT_WIDTH = 67;
     BOX_CONTENT_WIDTH
   );
 
+  // Build paths checked summary from the indicators (if available)
+  let pathsCheckedInfo = '';
+  if (cliAuthIndicators) {
+    const pathsChecked: string[] = [];
+
+    // Collect paths that were checked
+    if (cliAuthIndicators.checks.settingsFile.path) {
```
Comment (Shironex) [inline apps/server/src/index.ts]: > [!NOTE]
> **Nit - Codebase-fit:** Unnecessary truthy checks on always-populated path strings

The `path` field on `FileCheckResult` is typed as `string` (always truthy once set), and the check objects are always initialized with paths from the path helper functions. So `if (path)` will always be true. The conditional check is unnecessary but harmless - it just adds confusion about whether paths can be empty/undefined.

**Problematic code:**
```typescript
if (cliAuthIndicators.checks.settingsFile.path) {
  pathsChecked.push(`Settings: ${cliAuthIndicators.checks.settingsFile.path}`);
}
```

**Suggested fix:**
```diff
// The paths are always set, so the if-checks are unnecessary.
// Either remove them or add a comment explaining they're defensive:
pathsChecked.push(`Settings: ${cliAuthIndicators.checks.settingsFile.path}`);
pathsChecked.push(`Stats cache: ${cliAuthIndicators.checks.statsCache.path}`);
pathsChecked.push(`Projects dir: ${cliAuthIndicators.checks.projectsDir.path}`);
```
